cli/build.rs

```diff
@@ -25,7 +25,14 @@ fn read_git_sha() -> Option<String> {
         }
     }
 
-    let git_head_path = git_path.join("HEAD");
+    let git_dir_path;
+    if git_path.is_dir() {
+        git_dir_path = git_path;
+    } else {
+        git_dir_path = repo_path.join(fs::read_to_string(&git_path)
+                                      .unwrap().get(8..).unwrap().trim_end());
```

Comment:
Just for self-documentation purposes, could you use `"gitdir: ".len()` instead of `8` here?

Also, instead of using `unwrap` here, could you make the function return `None` if the `.git` file is unreadable?

This functionality (putting the sha in the version number) is so unimportant that I'd like the build to succeed even if some occurs when getting the sha.

Reply:
I've made both of these changes, although I'm definitely a Rust novice, so let me know if this is not the correct way of doing it.